src/core/osd.c

```diff
@@ -550,21 +690,28 @@ void osd_signal_update() {
 }
 
 void *thread_osd(void *ptr) {
+    static uint8_t fhd_d = 0;
     for (;;) {
         // wait for signal to render
         sem_wait(&osd_semaphore);
 
+        // clear shadow buffer when mode changes
+        if(fhd_d != is_fhd) {
+            for (int i = 0; i < HD_VMAX; i++) {
+                for (int j = 0; j < HD_HMAX; j++) osd_buf_shadow[i][j] = 0x20;
+            }
+            fhd_d = is_fhd;
+        }
+
+        // display osd
         for (int i = 0; i < HD_VMAX; i++) {
             for (int j = 0; j < HD_HMAX; j++) {
-                uint16_t ch = osd_buf[i][j];
-                if (ch == 0x20)
-                    ch = elrs_osd[i][j];
-                if (ch != osd_buf_shadow[i][j]) {
-                    osd_buf_shadow[i][j] = ch;
+                if (osd_buf[i][j] != osd_buf_shadow[i][j]) {
+                    osd_buf_shadow[i][j] = osd_buf[i][j];
```

Comment:
This change should be left as it was in the open source repo otherwise the ELRS backpack cannot add elements to the OSD.

Reply:
This PR is just for synchronizing SVN and Git repositories. If there is a bug, we can submit another PR after this.